ddtrace/internal/_threads.cpp

```diff
@@ -266,12 +275,14 @@ PeriodicThread_start(PeriodicThread* self, PyObject* args)
 
         bool error = false;
         auto interval = std::chrono::milliseconds((long long)(self->interval * 1000));
+        bool first_run = self->_no_wait_at_start;
```

Comment:
```suggestion
        if (self->_no_wait_at_start) self->_request.set();
```

?

would mean we don't need to keep a extra bool around and have the "complex" logic below for checking additional bool + setting to false on every iteration.

wdyt?

Reply:
yes, but 

> self->_request->wait(interval)

will sit and wait for up to interval for `self->_request` to be set, so we run at least once per-interval, unless we manually trigger for it to run.

if we do `self->_request->set()` then `self->_request->wait(interval)` will return true immediately, and then the loop will continue and the periodic function will execute immediately.



it should be exactly the same as you have it, just without needing to change the condition to also have to check/manage `first_run`

Reply:
Ok, I'll do the change and retest it tomorrow.
Hopefully we can merge it then.